src/utils/config_filters.hpp

```diff
@@ -69,4 +69,8 @@ bool string_matches_if_present(
 
 bool bool_or_empty(const config& filter, const config& cfg, const std::string& attribute);
 
+bool other_key_int(const config& cfg, const std::string& attribute);
+
+bool other_key_double(const config& cfg, const std::string& attribute);
```

Comment:
This is unreadable code, there's no reason to expect "int" to mean "is being used for addition/subtraction" or "double" to mean "is being used for multiplication/division". What I'd normally say in a code review is "this needs documentation, without just repeating the words used in the function name", in the expectation that a better name will be found in the process of writing the documentation.

However, IMO the functions have so little code in them that it should either be in `matches_ability_filter` or in a single utility function in the same file as `matches_ability_filter`, for example
```
/**
 * Returns true if any of "add", "sub", "multiply" or "divide" are present and have a value
 * other than their respective no-op value.
 */
has_value_changing_modifier(const config& cfg);

Reply:
it's done